core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -152,8 +159,19 @@ private[spark] class EventLoggingListener(
     }
   }
 
+  // We log the event both when stage submitted and stage completed, and after each logEvent call,
+  // replace the modifiedMetrics with the latestMetrics. In case the stages submit and complete
+  // time might be interleaved. So as to make the result the same with the running time.
+  private def logMetricsUpdateEvent() : Unit = {
+    modifiedMetrics.map(metrics => logEvent(metrics._2))
+    latestMetrics.map(metrics => modifiedMetrics.update(metrics._1, metrics._2))
+  }
```

Comment:
> I don't understand the last two sentences of the comment -- can you expand on that?

I'll update the code according to the design doc. I think the code is not that correct. Please refer it in [design doc](https://issues.apache.org/jira/secure/attachment/12762171/Tracking%20Spark%20Memory%20Usage%20-%20Phase%201.pdf)